src/Altinn.App.Api/Controllers/InstancesController.cs

```diff
@@ -535,22 +557,28 @@ public class InstancesController : ControllerBase
             {
                 return StatusCode((int)HttpStatusCode.Forbidden, validationResult);
             }
-
-            ProcessChangeContext processChangeContext = new(targetInstance, User)
+            
+            ProcessStartRequest processStartRequest = new()
             {
-                DontUpdateProcessAndDispatchEvents = true
+                Instance = targetInstance,
+                User = User,
+                Dryrun = true
             };
-            processChangeContext = await _processEngine.StartProcess(processChangeContext);
+            var startResult = await _processEngine.StartProcess(processStartRequest);
 
             targetInstance = await _instanceClient.CreateInstance(org, app, targetInstance);
 
             await CopyDataFromSourceInstance(application, targetInstance, sourceInstance);
 
             targetInstance = await _instanceClient.GetInstance(targetInstance);
 
-            processChangeContext.Instance = targetInstance;
-            processChangeContext.DontUpdateProcessAndDispatchEvents = false;
-            await _processEngine.StartTask(processChangeContext);
+            ProcessStartRequest rerunRequest = new()
+            {
+                Instance = targetInstance,
+                Dryrun = false,
+                User = User
+            };
+            await _processEngine.UpdateInstanceAndRerunEvents(rerunRequest, startResult.ProcessStateChange?.Events);
 
             await RegisterEvent("app.instance.created", targetInstance);
```

Comment:
What is the `app.instance.created` event compare to the `InstanceEventType.Created`? Should this method have been somewhere else? I'm thinking a EventDispatcherService class with a DispatchEvent method and avoid the direct use of the IEvents which ideally should have been renamed to IEventsClient (since it's a http client).

Reply:
The InstanceEventType.* is Enums for what events that can happen on a instance in the eyes of storage, see DispatchEventsToStorage. The string "app.instance.created" is the eventtype that is sent to the EventsComponent.
We could create an Enum for the static events dispatched to the eventcomponent but some of them have the taskId embedded in the eventID so we would still need to support dispatching events as string I think

Reply:
For string support we can create a static class with string constants avoiding repeating the string multiple places. Not a big thing in this case though.

"In the eyes of storge" - lets talk about that one:)